### PR TITLE
Common code refactor, back to mscalendar plugin

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Refactors the repository to be `mscalendar` only again.
- Moved all the common code to the `calendar` package where it could be used by others
- Refactored `server/mscalendar` into `calendar/engine`.
- Removed all Google Calendar code, assets, etc.

### CODE REVIEW

**Comment by fmartingr** [comment]
It seems that since I moved all common code it shows up as "new" so the lint is failing for all known issues that were already there. Will fix those in a bit.

**Comment by fmartingr** [comment]
Adding @lieut-data and @hanzei as additional reviewers to share the load. Not an in-depth review is needed, just making sure I didn't leave anything obvious.

**Review by mickmister — APPROVED** [review]
LGTM

File: msgraph/__debug_bin1974478355
[the remark this answers is not in the document]
Comment (fmartingr) [reply msgraph/__debug_bin1974478355:1]: Removed, good catch indeed!

File: build/custom.mk [hunk 1]
```diff
@@ -23,23 +22,23 @@ mock:
...
-	mockgen -destination server/mscalendar/mock_plugin_api/mock_plugin_api.go -package mock_plugin_api github.com/mattermost/mattermost-plugin-mscalendar/server/mscalendar PluginAPI
-	mockgen -destination server/remote/mock_remote/mock_remote.go github.com/mattermost/mattermost-plugin-mscalendar/server/remote Remote
-	mockgen -destination server/remote/mock_remote/mock_client.go github.com/mattermost/mattermost-plugin-mscalendar/server/remote Client
-	mockgen -destination server/utils/bot/mock_bot/mock_poster.go github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot Poster
-	mockgen -destination server/utils/bot/mock_bot/mock_admin.go github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot Admin
-	mockgen -destination server/utils/bot/mock_bot/mock_logger.go github.com/mattermost/mattermost-plugin-mscalendar/server/utils/bot Logger
-	mockgen -destination server/store/mock_store/mock_store.go github.com/mattermost/mattermost-plugin-mscalendar/server/store Store
+	mockgen -destination server/engine/mock_engine/mock_engine.go $(REPOSITORY_URL)/server/engine Engine
```
[the remark this answers is not in the document]
Comment (fmartingr) [reply build/custom.mk:25]: Totally!

**Pull request merged**